nservicebus/azure-service-bus/batching.md

```diff
@@ -0,0 +1,70 @@
+
+In order to increase messages throughput, the Azure Service Bus uses batching prior to writing to its internal store.  That involves delaying write operations. When batching is enabled, the write operations are delayed by 20 ms. Read operations are not affected. This helps to increase the throughput.  
+
+Batching store operations should be disabled for low throughput scenarios requiring low latency. 
+
+The batching is enabled by default. In order to disable it, set `EnableBatching` to `false` on [queues](/nservicebus/azure-service-bus/configuration/configuration.md#Queues), [topics](/nservicebus/azure-service-bus/configuration/configuration.md#Topics), and [subscriptions](/nservicebus/azure-service-bus/configuration/configuration.md#Subscriptions).
+
+
+## Client-side batching
+
+Client-side batching allows to delay sending messages to a queue or a topic for a short period of time. Additional messages sent to the same queue or topic within that period of time, they will be grouped and transmitted together in a single batch. This setting affects the Azure Service Bus Message Senders and can be configured using [`BatchFlushInterval`](/nservicebus/azure-service-bus/configuration/configuration.md#messaging-factories). Client-side batching is enabled by default.
+
+
+## Batching messages sent from a handler
+
+Azure Service Bus optimizes multiple message sends from a handler by batching send operations into a single operation. For example, when a handler is sending multiple messages, Azure Service Bus will batch those and send in as few operations as it can. Size of a batch cannot exceed the maximum size on a `BrokeredMessage`. The transport allows to configure what the maximum message size should be using [`MaximumMessageSizeInKilobytes`](/nservicebus/azure-service-bus/configuration/configuration.md#message-senders) setting of the Message Senders.
+
+NOTE: `BrokeredMessage` size is different between [tiers](https://azure.microsoft.com/en-us/documentation/articles/service-bus-premium-messaging/) of Azure Service Bus. 
+
+When batching messages sent from a handler, the underlying implementation of batching serializes messages. Serialized batch is usually bigger than the original messages combined. To ensure successful batch sending operation, batch should not exceed the `BrokeredMessage` size. To cater for the overhead caused by serialization, the final batch size is estimated using `MessageSizePaddingPercentage` setting. By default, it's set to 5%. It can be configured using [`MessageSizePaddingPercentage`](/nservicebus/azure-service-bus/configuration/configuration.md#message-senders) ` configuration of the Message Senders.
+
+By default, messages batch exceeding the maximum allowed size by Azure Service Bus, will throw `MessageTooLargeException`. Default behavior can by changed with [`OversizedBrokeredMessageHandler<T>(T)`](/nservicebus/azure-service-bus/configuration/configuration.md#message-senders) configuration of the Message Senders.
```

Comment:
`Default behavior can by changed with` = `The default behavior can by changed with`